jina/serve/gateway.py

```diff
@@ -176,6 +177,31 @@ async def shutdown(self):
         """Shutdown the server and free other allocated resources, e.g, streamer object, health check service, ..."""
         ...
 
+    async def warmup(self):
+        '''Run client._is_flow_ready() request to trigger the dry_run endpoint on each executor.
+        This forces the gateway to establish connection and open a gRPC channel to each executor so that the first
+        request doesn't need to experience the penalty of eastablishing a brand new gRPC channel.
+        '''
+        self.logger.debug('Running warmup')
+        from jina import Client
+
+        timeout = time.time() + 60 * 5  # 5 minutes from now
+
+        while True:
+            dry_run_responses = []
+            for port, protocol in zip(self.ports, self.protocols):
+                client = Client(
```

Comment:
we should not be using `Client`, we should be using the `streamer`. Maybe you can add a `warmup` to the streamer